api/logs/v1/labels_enforcer.go

```diff
@@ -131,3 +164,17 @@ func combineLabelMatchers(queryMatchers, authzMatchers []*labels.Matcher) []*lab
 
 	return matchers
 }
+
+func initAuthzMatchers(lm []*labels.Matcher) ([]*labels.Matcher, error) {
```

Comment:
Took me a while to understand what this function is doing, why it's recreating the same type. It seems like the answer is that this type was not meant to be serialized to a JSON representation and so is not properly initialized when de-serialized.

This is not really something for this PR, but I think introducing a custom type that makes it clear that the deserialized type can not actually be used as a Matcher should be cleaner.

Reply:
I agree but this requires refining the initial proposal (see ) and rework maybe opa-openshift and opa-ams sidecars:
https://github.com/observatorium/observatorium/blob/f174afa24b856b51b3575d45d8797503f3f22072/docs/proposals/drafts/20210709-opa-authorizer-labels-observatorium-api.md